### PR TITLE
Implement basic endpoints

### PR DESCRIPTION
**Pull request by ashleybudtke**
- endpoint to get all plants
- endpoint to get plant by id
- new controller and supporting domain

### CODE REVIEW

File: src/main/kotlin/com/github/ashleybudtke/happyplants/controllers/PlantController.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+class PlantController {
+
+    @Autowired
+    lateinit var plantService: PlantService
+
+    @GetMapping(value = ["/api/plants"])
```
Comment (ebudtke) [inline src/main/kotlin/com/github/ashleybudtke/happyplants/controllers/PlantController.kt:16]: Consider not hardcoding and moving to some kind of "paths" file, e.g. 
```suggestion
    @GetMapping(value = [PlantsPaths.BasePath])
```

File: src/main/kotlin/com/github/ashleybudtke/happyplants/controllers/PlantController.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    fun getAllPlants(): List<Plant> {
+        return plantService.getAllPlants()
+    }
+
+    @GetMapping(value = ["/api/plant/{id}"])
```
Comment (ebudtke) [inline src/main/kotlin/com/github/ashleybudtke/happyplants/controllers/PlantController.kt]: Same as above.

File: src/main/kotlin/com/github/ashleybudtke/happyplants/domain/PlantServiceImpl.kt [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package com.github.ashleybudtke.happyplants.domain
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+
+@Service
+class PlantServiceImpl : PlantService {
```
Comment (ebudtke) [inline src/main/kotlin/com/github/ashleybudtke/happyplants/domain/PlantServiceImpl.kt]: Why have this implement `PlantService`? Not sure the value in having that interface unless there's something coming soon.

Comment (ashleybudtke) [reply src/main/kotlin/com/github/ashleybudtke/happyplants/domain/PlantServiceImpl.kt]: Following SOLID principles out of habit, but I agree it's probably unnecessary for such a tiny project.

File: src/main/kotlin/com/github/ashleybudtke/happyplants/controllers/PlantController.kt (hunk 2 above)
Comment (ashleybudtke) [reply src/main/kotlin/com/github/ashleybudtke/happyplants/controllers/PlantController.kt]: Decided not to pull these out, but your comment pointed out that both mappings should have been `/api/plants` (plural) so I fixed that.

**Pull request merged**